modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/defragmentation/CachePartitionDefragmentationManager.java

```diff
@@ -593,7 +590,13 @@ private boolean defragmentOnePartition(
 
     /** */
     public IgniteInternalFuture<?> completionFuture() {
-        return completionFut.chain(future -> null);
+        return completionFut.chain(future -> {
```

Comment:
What if no one's calling this method? I'd move chaining into future creation instead